src/celeritas/neutron/model/detail/NuclearZoneBuilder.hh

```diff
@@ -242,162 +180,153 @@ void NuclearZoneBuilder::calc_zone_component(IsotopeView const& target,
- * \f[
-     R = [ 1.16 * A^{1/3} - 1.3456 / A^{1/3} ] \cdot R_{scale}
-   \f]
- * where \f$ R_{scale} \f$ is a configurable parameter in [femtometer], while
- * \f$ R = 1.2 A^{1/3} \cdot R_{scale} \f$ (default) with a single parameter.
+ * Lightweight nuclei are treated as simple balls.
  */
-real_type NuclearZoneBuilder::calc_nuclear_radius(AtomicMassNumber a)
+auto NuclearZoneBuilder::calc_zones_light(AtomicMassNumber a) const
+    -> VecZoneDensity
 {
-    // Nuclear radius computed from A
-    real_type nuclear_radius{0};
-    auto amass = a.get();
-
-    if (amass > 4)
-    {
-        real_type cbrt_a = std::cbrt(static_cast<real_type>(amass));
-        real_type par_a = (options_.use_two_params ? 1.16 : 1.2);
-        real_type par_b = (options_.use_two_params ? -1.3456 : 0);
-        nuclear_radius = options_.radius_scale
-                         * (par_a * cbrt_a + par_b / cbrt_a);
-    }
-    else
-    {
-        nuclear_radius = options_.radius_small
-                         * (amass == 4 ? options_.radius_alpha : 1);
-    }
-
-    return nuclear_radius;
+    CELER_EXPECT(a <= AtomicMassNumber{4});
+    ZoneDensity result;
+    result.radius = options_.radius_small
+                    * (a == AtomicMassNumber{4} ? options_.radius_alpha : 1);
+    result.integral = 1;
+    return {result};
 }
 
 //---------------------------------------------------------------------------//
 /*!
- * Integrate the Woods-Saxon potential, \f$ V(r) \f$,
- *
- * \f[
-     V(r) = frac{V_{o}}{1 + e^{\frac{r - R}{a}}}
-   \f]
- * numerically over the volume from \f$ r_{min} \f$ to \f$ r_{rmax} \f$, where
- * \f$ V_{o}, R, a\f$ are the potential well depth, nuclear radius, and
- * surface thickness (skin depth), respectively.
+ * Small nuclei have a three-zone gaussian potential.
  */
-real_type
-NuclearZoneBuilder::integrate_woods_saxon(real_type rmin,
-                                          real_type rmax,
-                                          real_type nuclear_radius) const
+auto NuclearZoneBuilder::calc_zones_small(AtomicMassNumber a) const
+    -> VecZoneDensity
 {
-    real_type skin_ratio = nuclear_radius / skin_depth_;
+    real_type const nuclear_radius = this->calc_nuclear_radius(a);
+    real_type gauss_radius = std::sqrt(
+        ipow<2>(nuclear_radius) * (1 - 1 / static_cast<real_type>(a.get()))
+        + real_type{6.4});
 
-    real_type ws_shift = 2 * skin_ratio;
-    real_type delta_r = rmax - rmin;
+    Integrator integrate_gauss{
+        [](real_type r) { return ipow<2>(r) * std::exp(-ipow<2>(r)); }};
 
-    // Average value of \f$ r^{2} V(r) dr \f$ at boundaries
-    real_type integral = half() * delta_r
-                         * (rmin * (rmin + ws_shift) / (1 + std::exp(rmin))
-                            + rmax * (rmax + ws_shift) / (1 + std::exp(rmax)));
+    // Precompute y = sqrt(-log(alpha)) where alpha[3] = {0.7, 0.3, 0.01}
+    real_type const y[] = {0.597223, 1.09726, 2.14597};
```

Comment:
It occurs to me that since the function we're integrating uses fixed intervals and is constant with respect to the nucleus, we could simply precompute the gaussian integral...